sentry/src/test/java/io/sentry/SentryTracerTest.kt

```diff
@@ -562,4 +566,102 @@ class SentryTracerTest {
             anyOrNull()
         )
     }
+
+    @Test
+    fun `when initialized without idleTimeout, does not schedule finish timer`() {
+        val transaction = fixture.getSut()
+
+        assertNull(transaction.timerTask)
+    }
+
+    @Test
+    fun `when initialized with idleTimeout, schedules finish timer`() {
+        val transaction = fixture.getSut(idleTimeout = 50)
+
+        assertNotNull(transaction.timerTask)
+    }
+
+    @Test
+    fun `when no children and the transaction is idle, drops the transaction`() {
+        val latch = CountDownLatch(1)
+        fixture.getSut(idleTimeout = 10, latch = latch)
+
+        latch.await()
+
+        verify(fixture.hub, never()).captureTransaction(
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+        )
+    }
+
+    @Test
+    fun `when idle transaction with children, finishes the transaction after the idle timeout`() {
+        val latch = CountDownLatch(1)
+        val transaction = fixture.getSut(waitForChildren = true, idleTimeout = 10, latch = latch)
+
+        val span = transaction.startChild("op")
+        span.finish()
+
+        latch.await()
+
+        verify(fixture.hub).captureTransaction(
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+        )
+    }
+
+    @Test
+    fun `when a new child is added to the idle transaction, cancels the timer`() {
+        val transaction = fixture.getSut(waitForChildren = true, idleTimeout = 3000)
+
+        transaction.startChild("op")
+
+        assertNull(transaction.timerTask)
+    }
+
+    @Test
+    fun `when a child is finished and the transaction is idle, resets the timer`() {
+        val transaction = fixture.getSut(waitForChildren = true, idleTimeout = 3000)
+
+        val initialTime = transaction.timerTask!!.scheduledExecutionTime()
+
+        val span = transaction.startChild("op")
+        Thread.sleep(10)
```

Comment:
mm, we need a sleep here for sure, otherwise the code might be too quick to execute, so there's no actual delay in between 2 timer schedules. but I believe we can make just 1ms and it still should work + the impact is gonna be almost unnoticeable